framework/src/media/audio/audio_manager.c

```diff
@@ -159,18 +160,30 @@ static audio_manager_result_t set_audio_volume(audio_card_type_t type, uint8_t v
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
-static audio_manager_result_t find_audio_card_if_empty(audio_card_type_t card_type)
+static audio_manager_result_t init_audio_stream(audio_card_type_t card_type)
 {
-	int id = g_actual_audio_out_card_id;
-
-	if (card_type == INPUT) {
-		id = g_actual_audio_in_card_id;
+	if (((card_type == INPUT) && (g_actual_audio_in_card_id >= 0)) || ((card_type == OUTPUT) && (g_actual_audio_out_card_id >= 0))) {
```

Comment:
This condition means we found card already, so we can skip find_audio_card & get_actual_audio_xx_card_id.

Reply:
Once cards are found, just "find_audio_card" will be skipped by 'goto' at line 166.
And "get_actual_audio_xx_card_id" can also be called correctly by the 'goto' after cards are found.
So.. there may be no problem, I think..